components/ratgdo/ratgdo.cpp

```diff
@@ -144,29 +146,67 @@ namespace ratgdo {
                 }
             }
 
-            if (door_state == DoorState::OPEN) {
-                this->door_position = 1.0;
-            } else if (door_state == DoorState::CLOSED) {
-                this->door_position = 0.0;
-            } else {
-                if (*this->closing_duration == 0 || *this->opening_duration == 0 || *this->door_position == DOOR_POSITION_UNKNOWN) {
-                    this->door_position = 0.5; // best guess
+
+            if (door_state == DoorState::OPENING) {
+                // door started opening
+                if (prev_door_state == DoorState::CLOSING) {
+                    this->door_position_update();
+                    this->cancel_position_sync_callbacks();
+                    this->door_move_delta = DOOR_DELTA_UNKNOWN;
                 }
-            }
+                this->door_start_moving = millis();
+                this->door_start_position = *this->door_position;
+                if (this->door_move_delta == DOOR_DELTA_UNKNOWN) {
+                    this->door_move_delta = 1.0 - this->door_start_position;
+                }
+                this->schedule_door_position_sync();
 
-            if (door_state == DoorState::OPENING && !this->moving_to_position) {
-                this->position_sync_while_opening(1.0 - *this->door_position);
-                this->moving_to_position = true;
+                // this would only get called if no status message is received after door stops moving
+                // request a status message in that case
+                set_timeout("door_status_update", (*this->opening_duration + 1) * 1000, [=]() {
+                    this->send_command(Command::GET_STATUS);
+                });
             }
-            if (door_state == DoorState::CLOSING && !this->moving_to_position) {
-                this->position_sync_while_closing(*this->door_position);
-                this->moving_to_position = true;
+
+            if (door_state == DoorState::CLOSING) {
```

Comment:
Looks like these could be else if blocks

Reply:
Oh, yeah, I was planing to update them to `else if`...